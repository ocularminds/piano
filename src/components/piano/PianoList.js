import React from 'react';
import Piano from './Piano';

class PianoList extends React.Component {
  constructor(props) {
    super(props);
    this.createPianos = this.createPianos.bind(this);
  }

  createPianos(index, piano) {
    return (
      <Piano key={index} model={piano} onPlay={this.props.onPlay} id={index} />
    );
  }

  render() {
    return (
      <div>
        <h1>-- My Piano --</h1>
        {this.props.pianos &&
          this.props.pianos.map((piano, index) =>
            this.createPianos(index, piano)
          )}
      </div>
    );
  }
}

export default PianoList;
