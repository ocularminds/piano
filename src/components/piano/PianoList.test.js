import React from 'react';
import {render, getByText} from '@testing-library/react';
import PianoList from './PianoList';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

describe('Test for PianoList contents when rendered', () => {
  const pianos = [
    {
      name: 'Casino',
      keyPads: ['C', 'D', 'E', 'F', 'G', 'A', 'B', 'C'],
    },
    {
      name: 'Korg 7781',
      keyPads: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'A'],
    },
  ];
  
  it('should match the snapshot', () => {
    const play = (e) => {};
    const wrapper = shallow(<PianoList pianos={pianos} onPlay={play} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders My Piano as Header', () => {
    const {getByText} = render(<PianoList />);
    const linkElement = getByText(/My piano/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('Should contain piano name in the properties', () => {
    const play = (e) => {};
    const wrapper = shallow(<PianoList pianos={pianos} onPlay={play} />);
    const result = wrapper.instance().createPianos(2, pianos[1]);
    expect(result['props'].model.name).toEqual('Korg 7781');
  });

  it('Should contain total number of pianos when render', () => {
    const play = (e) => {};
    const wrapper = shallow(<PianoList pianos={pianos} onPlay={play} />);
    const result = wrapper.instance().render();
    expect(result['props'].children.length).toEqual(2);
  });
});
