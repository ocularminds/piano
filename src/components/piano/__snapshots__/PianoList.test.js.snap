// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for PianoList contents when rendered should match the snapshot 1`] = `ShallowWrapper {}`;
